### ISSUE TITLE
GCC 10:  error: ‘strcasecmp’ was not declared in this scope

### ISSUE THREAD

**Issue by MBeijer** [sba1/adtools#110]
Hi!

I'm having trouble compiling a program since I updated to GCC 10, where I get the error
```c++
 error: ‘strcasecmp’ was not declared in this scope
```
even though I have these includes:
```c++
    #include <cstdio>
    #include <cstdlib>
    #include <cstring>
    #include <strings.h>
```
or these:
```c++
    #include <stdio.h>
    #include <stdlib.h>
    #include <string.h>
    #include <strings.h>
```

with `-mcrt=newlib`

I looked at the newlib includes and they seem to define the function, so it seems like it's not loading the proper includes from newlib for some reason.

I hope someone can help out with this!

**Comment by sodero** [comment]
How did you build the compiler? Are you cross compiling or building natively? What happens if you use `-mcrt=clib2` instead?

**Comment by sodero** [comment]
Are you using `using namespace std;` / `std::strcasecmp` btw?

**Comment by MBeijer** [comment]
@sodero no, I'm not using `using namespace std;` nor `std::strcasecmp` but I tried using that now and I get this error instead:
```
error: ‘strcasecmp’ is not a member of ‘std’; did you mean ‘strcasecmp’?
```

:sweat_smile:

**Comment by sodero** [comment]
Does this work for you?

![bild](https://user-images.githubusercontent.com/1541697/148586558-956be55d-7a90-4d3b-8931-50c5fdf01666.png)

Sorry about the weird way of showing example code. I have networking issues in my Amiga cave.

**Comment by sodero** [comment]
I would recommend using gcc 11 and clib2 btw. If you're using a native compiler you could pick these, [gcc 11](https://github.com/sodero/adtools/releases/download/10.3.0_1/adtools-os4-gcc11.1.0-20210531-743.lha) and
[clib2](https://github.com/sodero/adtools/releases/download/10.3.0_1/adtools-os4-gcc11.1.0_clib2-20210531-743.lha).

**Comment by MBeijer** [comment]
I'm cross compiling from Linux. Will do some tests with your example code
and get back to you with the results.

**Comment by MBeijer** [comment]
Strange, with that test code it worked fine with `-athread=native
-mcrt=newlib` and `-athread=single -mcrt=clib2`

It's really strange that the code I compile to get the error has the same
includes +  `-athread=native -mcrt=newlib`.
The same code compiles just fine with both the MorphOS cross toolchain and
Bebbo's m68k toolchain without issue.

**Comment by capehill** [comment]
In case you were building with -std=c++11 (or later), try -std=gnu++11. In newlib, strcasecmp and some other functions are not available if __STRICT_ANSI__ is defined.

I have seen similar problem when building MilkyTracker. First I tried -U__STRICT_ANSI__ which works around it, but gnu++11 is probably a cleaner solution.

**Comment by MBeijer** [comment]
Ah, okay, this was probably when trying to compile MilkyTracker... Can't
remember.
